package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -0,0 +1,943 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+#
+# Copyright (C) 2013-2018  Sébastien Buchoux <sebastien.buchoux@gmail.com>
+# Copyright (c) 2018 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v3 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+#
+
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: initializedcheck=False
+
+"""
+Neighbor search library --- :mod:`MDAnalysis.lib.grid`
+======================================================
+
+This Neighbor search library is a serialized Cython port of the NS grid search implemented in GROMACS.
+"""
+
+
+# Preprocessor DEFs
+DEF DIM = 3
+DEF XX = 0
+DEF YY = 1
+DEF ZZ = 2
+
+DEF EPSILON = 1e-5
+
+DEF BOX_MARGIN=1.0010
+DEF MAX_NTRICVEC=12
+
+# Used to handle memory allocation
+from cpython.mem cimport PyMem_Malloc, PyMem_Realloc, PyMem_Free
+from libc.math cimport sqrt
+import numpy as np
+cimport numpy as np
+
+ctypedef np.int_t ns_int
+ctypedef np.float32_t real
+ctypedef real rvec[DIM]
+ctypedef ns_int ivec[DIM]
+ctypedef ns_int ipair[2]
+ctypedef real matrix[DIM][DIM]
+
+
+# Useful Functions
+cdef real rvec_norm2(const rvec a) nogil:
+    return a[XX]*a[XX]+a[YY]*a[YY]+a[ZZ]*a[ZZ]
+
+cdef void rvec_clear(rvec a) nogil:
+    a[XX]=0.0
+    a[YY]=0.0
+    a[ZZ]=0.0
+
+########################################################################################################################
+#
+# Utility class to handle PBC
+#
+########################################################################################################################
+cdef struct cPBCBox_t:
+    matrix     box
+    rvec       fbox_diag
+    rvec       hbox_diag
+    rvec       mhbox_diag
+    real       max_cutoff2
+    ns_int        ntric_vec
+    ns_int[DIM]   tric_shift[MAX_NTRICVEC]
+    real[DIM]  tric_vec[MAX_NTRICVEC]
+
+
+# Class to handle PBC calculations
+cdef class PBCBox(object):
+    cdef cPBCBox_t c_pbcbox
+    cdef rvec center
+    cdef rvec bbox_center
+    cdef bint is_triclinic
+
+    def __init__(self, real[:,::1] box):
+        self.update(box)
+
+    cdef void fast_update(self, real[:,::1] box) nogil:

```

Comment:
In general, because this isn't a parallel module, all references to gil can be removed right?

Reply:
I would leave the GIL code in – you never know under which circumstances someone is trying to run the code and having the GIL/NOGIL here seems like valuable information, i.e., someone has thought about the code. Seems a shame to throw this knowledge away.